src/sdl2/hint.rs

```diff
@@ -74,6 +75,32 @@ pub fn get_video_minimize_on_focus_loss() -> bool {
     )
 }
 
+/// A hint that overrides the user's locale settings.
+///
+/// [Official SDL documentation](https://wiki.libsdl.org/SDL2/SDL_HINT_PREFERRED_LOCALES)
+///
+/// # Default
+/// This is disabled by default.
+///
+/// # Example
+///
+/// See [`crate::locale::get_preferred_locales`].
+pub fn set_preferred_locales<T: std::borrow::Borrow<Locale>>(
+    locales: impl IntoIterator<Item = T>,
+) -> bool {
+    set(PREFERRED_LOCALES, &format_locale_hint(locales))
+}
+
+fn format_locale_hint<T: std::borrow::Borrow<Locale>>(
+    locales: impl IntoIterator<Item = T>,
+) -> String {
+    locales
+        .into_iter()
+        .map(|locale| locale.borrow().to_string())
+        .collect::<Vec<_>>()
+        .join(",")
```

Comment:
There are a lot of unnecessary heap allocations made here, you can allocate one String (using `with_capacity` with a good guess, like size hint lower bound * 6) and write to that using `write!`